src/me/mrCookieSlime/Slimefun/listeners/FurnaceListener.java

```diff
@@ -22,24 +22,19 @@ public FurnaceListener(SlimefunStartup plugin) {
 	
 	@EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
 	public void onBurn(FurnaceBurnEvent e) {
-		if (BlockStorage.check(e.getBlock()) != null && BlockStorage.check(e.getBlock()) instanceof EnhancedFurnace) {
-			EnhancedFurnace furnace = (EnhancedFurnace) BlockStorage.check(e.getBlock());
-			if (furnace.getFuelEfficiency() > 0) e.setBurnTime(((int) ((1.0 + 0.2 * furnace.getFuelEfficiency()) * e.getBurnTime())));
-		}
+		EnhancedFurnace furnace = (EnhancedFurnace) BlockStorage.check(e.getBlock());
+		if (furnace instanceof EnhancedFurnace && furnace.getFuelEfficiency() > 0)
+			e.setBurnTime(((int) ((1 + 0.2 * furnace.getFuelEfficiency()) * e.getBurnTime())));
 	}
 
 	@EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
 	public void onSmelt(FurnaceSmeltEvent e) {
-		if (BlockStorage.check(e.getBlock()) != null && BlockStorage.check(e.getBlock()) instanceof EnhancedFurnace) {
-			EnhancedFurnace furnace = (EnhancedFurnace) BlockStorage.check(e.getBlock());
+		EnhancedFurnace furnace = (EnhancedFurnace) BlockStorage.check(e.getBlock());
+		if (furnace instanceof EnhancedFurnace) {
 			Furnace f = (Furnace) e.getBlock().getState();
-			int amount = f.getInventory().getSmelting().getType().toString().endsWith("_ORE") ? furnace.getOutput(): 1;
-			ItemStack output = RecipeCalculator.getSmeltedOutput(f.getInventory().getSmelting().getType());
-			ItemStack result = f.getInventory().getResult();
-			if (result != null) result = result.clone();
-			f.getInventory().setResult(null);
-			if (result != null) f.getInventory().setResult(new CustomItem(result, result.getAmount() + amount > result.getMaxStackSize() ? result.getMaxStackSize(): result.getAmount() + amount));
-			else f.getInventory().setResult(new CustomItem(output, output.getAmount() + amount > output.getType().getMaxStackSize() ? output.getType().getMaxStackSize(): output.getAmount() + amount));
+			int amount = f.getInventory().getSmelting().getType().toString().endsWith("_ORE") ? furnace.getOutput() : 1;
+			ItemStack result = f.getInventory().getResult() == null ? RecipeCalculator.getSmeltedOutput(f.getInventory().getSmelting().getType()) : f.getInventory().getResult().clone();
```

Comment:
The error would probably still persist, as the error came from "getSmeltedOutput" returning null.
A null-check after storing the output like before would be better.

Maybe getSmeltedOutput() is not working properly anymore and that is the cause of this problem.